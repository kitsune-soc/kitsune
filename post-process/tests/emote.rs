use pest::Parser;
use post_process::{PostParser, Rule};
use pretty_assertions::assert_eq;

#[test]
fn parse_emote() {
    let text = "hello :blobfoxcoffee: how are you doing?:blobcatpeek:";
    let mut token_iter = PostParser::parse(Rule::post, text).expect("Failed to parse post");

    let text1 = token_iter.next().unwrap();
    assert_eq!(text1.as_rule(), Rule::text);
    assert_eq!(text1.as_str(), "hello ");

    let emote1 = token_iter.next().unwrap();
    assert_eq!(emote1.as_rule(), Rule::emote);
    assert_eq!(emote1.as_str(), ":blobfoxcoffee:");
    let mut emote1 = emote1.into_inner();
    let emote1_content = emote1.next().unwrap();
    assert_eq!(emote1_content.as_rule(), Rule::emote_content);
    assert_eq!(emote1_content.as_str(), "blobfoxcoffee");

    let text2 = token_iter.next().unwrap();
    assert_eq!(text2.as_rule(), Rule::text);
    assert_eq!(text2.as_str(), " how are you doing?");

    let emote2 = token_iter.next().unwrap();
    assert_eq!(emote2.as_rule(), Rule::emote);
    assert_eq!(emote2.as_str(), ":blobcatpeek:");
    let mut emote2 = emote2.into_inner();
    let emote2_content = emote2.next().unwrap();
    assert_eq!(emote2_content.as_rule(), Rule::emote_content);
    assert_eq!(emote2_content.as_str(), "blobcatpeek");
}
