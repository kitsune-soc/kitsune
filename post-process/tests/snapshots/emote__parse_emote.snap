---
source: post-process/tests/emote.rs
expression: "PostParser::parse(Rule::post, &post).unwrap()"
input_file: post-process/tests/input/emote/full_post_1
---
[
    Pair {
        rule: text,
        span: Span {
            str: "hello ",
            start: 0,
            end: 6,
        },
        inner: [],
    },
    Pair {
        rule: emote,
        span: Span {
            str: ":blobfoxcoffee:",
            start: 6,
            end: 21,
        },
        inner: [
            Pair {
                rule: emote_content,
                span: Span {
                    str: "blobfoxcoffee",
                    start: 7,
                    end: 20,
                },
                inner: [],
            },
        ],
    },
    Pair {
        rule: text,
        span: Span {
            str: " how are you doing?",
            start: 21,
            end: 40,
        },
        inner: [],
    },
    Pair {
        rule: emote,
        span: Span {
            str: ":blobcatpeek:",
            start: 40,
            end: 53,
        },
        inner: [
            Pair {
                rule: emote_content,
                span: Span {
                    str: "blobcatpeek",
                    start: 41,
                    end: 52,
                },
                inner: [],
            },
        ],
    },
]
